src/OpenTelemetry.Api/Context/RuntimeContext.cs

```diff
@@ -67,7 +67,12 @@ public static RuntimeContextSlot<T> RegisterSlot<T>(string name)
         /// <returns>The slot previously registered.</returns>
         public static RuntimeContextSlot<T> GetSlot<T>(string name)
         {
-            return (RuntimeContextSlot<T>)Slots[name];
+            if (Slots.TryGetValue(name, out var slot) && slot is RuntimeContextSlot<T> expectedSlot)
+            {
+                return expectedSlot;
+            }
+
+            throw new Exception("Slot with given name not found");

```

Comment:
```suggestion
```suggestion
            Slots.TryGetValue(name, out var slot);
            return slot as RuntimeContextSlot<T> ?? throw new Exception("Slot with given name not found");
```

Reply:
And we probably want to use `KeyNotFoundException` instead of a general one?

Reply:
yes. agree

Reply:
E.g. `throw new KeyNotFoundException(name)`.

BTW, we probably also need to check if key is null or empty string.